nifi-nar-bundles/nifi-framework-bundle/nifi-stateless-bundle/README.md

```diff
@@ -0,0 +1,405 @@
+going to succeed the next time around. Such a case may result in constantly consuming the same data and attempting to process it over and over again. However, it may make sense if the use case
+dictates that no more data may be processed until such a message on the Kafka queue has been properly dealt with. 
+
+##### Full Examples
+
+An example of a fully formed dataflow configuration file that will import a dataflow from NiFi Registry is as follows:
+```
+nifi.stateless.registry.url=https://nifi-registry/
+nifi.stateless.flow.bucketId=00000000-0000-0000-0000-000000000011
+nifi.stateless.flow.id=00000000-0000-0000-0000-000000000044
+nifi.stateless.flow.version=5
+
+nifi.stateless.parameters.kafkahdfs.Kafka to HDFS
+nifi.stateless.parameters.kafkahdfs.Kafka Topic=Sensor Data
+nifi.stateless.parameters.kafkahdfs.Kafka Brokers=kafka-01:9092,kafka-02:9092,kafka-03:9093
+nifi.stateless.parameters.kafkahdfs.HDFS Directory=/data/sensors
+
+nifi.stateless.reporting.task.stats.name=Stats
+nifi.stateless.reporting.task.stats.type=ControllerStatusReportingTask
+nifi.stateless.reporting.task.stats.bundle=org.apache.nifi:nifi-standard-nar:1.12.1
+nifi.stateless.reporting.task.stats.properties.Show Deltas=false
+nifi.stateless.reporting.task.stats.frequency=1 minute
+nifi.stateless.reporting.task.stats.properties.Reporting Granularity=One Second
+```
+
+An alternate example, referencing a locally stored JSON file for the dataflow:
+```
+nifi.stateless.flow.snapshot.file=/var/lib/nifi/stateless-flows/kafka-to-hdfs.json
+
+nifi.stateless.parameters.kafkahdfs.Kafka to HDFS
+nifi.stateless.parameters.kafkahdfs.Kafka Topic=Sensor Data
+nifi.stateless.parameters.kafkahdfs.Kafka Brokers=kafka-01:9092,kafka-02:9092,kafka-03:9093
+nifi.stateless.parameters.kafkahdfs.HDFS Directory=/data/sensors
+
+nifi.stateless.reporting.task.stats.name=Stats
+nifi.stateless.reporting.task.stats.type=ControllerStatusReportingTask
+nifi.stateless.reporting.task.stats.bundle=org.apache.nifi:nifi-standard-nar:1.12.1
+nifi.stateless.reporting.task.stats.properties.Show Deltas=false
+nifi.stateless.reporting.task.stats.frequency=1 minute
+nifi.stateless.reporting.task.stats.properties.Reporting Granularity=One Second
+```
+
+
+An example of a minimal configuration with no Reporting Tasks or Parameters:
+```
+nifi.stateless.flow.snapshot.file=/var/lib/nifi/stateless-flows/kafka-to-hdfs.json
+```
+
+
+#### Passing Parameters
+
+There are times when it is not convenient to pass Parameters in the .properties file.
+For example, the same dataflow may be reused for several different sources or sinks, and each time the flow is run,
+it needs to be run with a different set of Parameters.
+
+Additionally, there may be sensitive parameters that users prefer not to include in the .properties file. These may be provided via
+Environment Variables, for example.
```

Comment:
Describing this approach to passing parameters as _Environment Variables_ could be confusing since the examples show command line arguments, and the BootstrapConfiguration class parses command line arguments.  On the other hand, enhancing the BootstrapConfiguration class to parse parameters provided as environment variables accessible through System.getenv() would be a great feature.  Perhaps support for System environment variables could be implemented later and the documentation adjusted for now.

Reply:
I would agree that the statement here should probably say:

_"These may be provided via command line arguments"_

If you specified an environment variable as part of the argument, would it get substituted somewhere along the way in the shell script that is executing stateless, or would it need special handling like David mentioned?

`-p "Kafka Parameter Context:Kafka Topic:${KAFKA_SENSOR_TOPIC}"`